### PR TITLE
Auth Register Functionality

### PR DESCRIPTION
**Pull request by RichardHpa**

### CODE REVIEW

File: controllers/users.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+function randomTokenString() {
+  return crypto.randomBytes(40).toString('hex');
+}
+
+async function sendVerificationEmail(account: User, origin: string) {
+  let message;
+  if (origin) {
+    const verifyUrl = `${origin}/verify-email?token=${account.verificationToken}`;
+    message = `<p>Please click the below link to verify your email address:</p>
+                 <p><a href="${verifyUrl}" target="_blank">${verifyUrl}</a></p>`;
```
Comment (jenniferexong) [inline controllers/users.ts:73]: Do we need this here?
```suggestion
                 <p><a href="${verifyUrl}" target="_blank" rel="noreferrer noopener">${verifyUrl}</a></p>`;
```

**Review by jenniferexong — APPROVED** [review]
With my very very very limited knowledge of this area, lgtm apart from two suggestions!

File: controllers/users.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+require('dotenv').config();
+import type { Request, Response } from 'express';
+import { Router } from 'express';
+import { Users } from '../models/Users';
+import { roles } from '../_helpers/roles';
+import bcrypt from 'bcryptjs';
+import { sendEmail } from '../_helpers/sendEmail';
```
Comment (jenniferexong) [inline controllers/users.ts:7]: Maybe absolute imports?

File: controllers/users.ts (hunk 1 above)
Comment (RichardHpa) [reply controllers/users.ts:73]: Not really for this app but its just a standard thing you add if you are doing password verification

**Pull request merged**